identity-did/src/verification/method.rs

```diff
@@ -132,3 +132,39 @@ impl<T> AsRef<DID> for Method<T> {
     self.id()
   }
 }
+
+/// Represents all possible verification method types
+///
+/// see [W3C DID-core spec](https://www.w3.org/TR/did-core/#relative-did-urls)
+pub enum MethodUriType {
+  Absolute,
+  Relative,
+}
+/// Used to return absolute or relative method URI.
+///
+/// This trait is used to determine whether absolute or relative method URIs
+/// should be used to signing data.
+///
+/// [More Info](https://www.w3.org/TR/did-core/#relative-did-urls)
+pub trait TryMethod {
+  /// Flag that determines whether absolute or rleative URI
+  const TYPE: MethodUriType;
+  /// Returns String representation of absolute or relative method URI, if any.
+  fn method(method_id: &DID) -> Option<String> {
+    let fragment = method_id.fragment()?;
+
+    Some(match Self::TYPE {
+      MethodUriType::Absolute => method_id.to_string(),
+      MethodUriType::Relative => core::iter::once('#').chain(fragment.chars()).collect(),
```

Comment:
if you change the argument to `method: &Method<T>` you won't have to re-implement `try_into_fragment` 😉